[Excel] (Custom Functions) Further explain custom function limitations (#3610)
OfficeDev/office-js-docs-pr: docs/excel/call-excel-apis-from-custom-function.md

## Limitations of calling Excel JavaScript APIs through a custom function

> A custom functions add-in has the ability to call Excel JavaScript APIs, but you should be cautious about which Excel JavaScript APIs your add-in calls. Don't call Excel JavaScript APIs from a custom function that change cells outside of the cell running the custom function. Changing other cells or the Excel environment can result in poor performance, time outs, and infinite loops in the Excel application. This means your custom functions shouldn't do any of the following.

Reviewer: Suggestions for conciseness.
```suggestion
A custom functions add-in can call Excel JavaScript APIs, but you should be cautious about which APIs it calls. Don't call Excel JavaScript APIs from a custom function that change cells outside of the cell running the custom function. Changing other cells or the Excel environment can result in poor performance, time outs, and infinite loops in the Excel application. This means your custom functions shouldn't do any of the following.
```

- Insert, delete, or format cells on the spreadsheet.
- Change another cell's value.
- Move, rename, delete, or add sheets to a workbook.
- Add names to a workbook.
- Set properties.
- Change any of the Excel environment options, such as calculation mode or screen views.

> Your custom functions add-in can retrieve information from cells outside the cell running the custom function, but it shouldn't perform write operations to other cells. Instead, make changes to other cells or to the Excel environment from the context of a ribbon button or a task pane. In addition, custom function calculations shouldn't run while an Excel recalculation is taking place, as this scenario creates unpredictable results.

Reviewer: I like using the read/write terminology here to reinforce the underlying actions. Also, conciseness suggestion.
```suggestion
Your custom functions add-in can read information from cells outside the cell running the custom function, but it shouldn't perform write operations to other cells. Instead, make those changes from the context of a ribbon button or a task pane. In addition, custom function calculations shouldn't run while an Excel recalculation is taking place, as this scenario creates unpredictable results.
```

Author: Thanks for the review and suggestions! I'll take the read/write half of this one, but retain the "changes to other cells or to the Excel environment" phrase to reinforce that the instruction to use a task pane or ribbon button applies to both changes to other cells or to Excel environment.
